src/ripple/app/main/Application.cpp

```diff
@@ -2006,6 +2061,32 @@ bool ApplicationImp::updateTables ()
     return true;
 }
 
+bool ApplicationImp::validateShards()
+{
+    if (!m_overlay)
+        Throw<std::runtime_error>("no overlay");
+    if(config_->standalone())
+    {
+        JLOG(m_journal.fatal()) <<
+            "Shard validation cannot be run in standalone";
+        return false;
+    }
+    if (config_->section(ConfigSection::shardDatabase()).empty())
+    {
+        JLOG (m_journal.fatal()) <<
+            "The [shard_db] configuration setting must be set";
+        return false;
+    }
+    if (!shardStore_)
+    {
+        JLOG(m_journal.fatal()) <<
+            "Invalid [shard_db] configuration";
+        return false;
+    }
+    shardStore_->validate();
```

Comment:
I am not sure a boolean return would be very useful. One or more shards may fail to validate and that is logged but it is up to the user to take action. In the unlikely event that a shard is corrupt, they must delete the shard, same as the node store. If it is missing a node (should never happen on a complete shard), it will auto heal through SHAMap triggering the acquire of the missing nodes.